### PR TITLE
Add rules_ios 3.0.0

### PR DESCRIPTION
**Pull request by luispadron**

### CODE REVIEW

File: modules/rules_ios/3.0.0/presubmit.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+bcr_test_module:
+  module_path: "e2e/bzlmod"
+  matrix:
+    platform: ["macos"]
+  tasks:
```
Comment (fmeum) [inline modules/rules_ios/3.0.0/presubmit.yml:5]: You need to have a named task as a layer underneath.

Comment (luispadron) [reply modules/rules_ios/3.0.0/presubmit.yml]: Updated this, was confused when the tool asked for a `presubmit.yml`, I gave it the path to the rules_ios one but this should just be set to the name of that task from what i can see in other PRs

File: modules/rules_ios/3.0.0/MODULE.bazel [hunk 2]
```diff
@@ -0,0 +1,125 @@
+"""
+Defines all the external repositories and dependencies for rules_ios.
+"""
+
+# Defines the rules_ios bzlmod module.
+# Version is updated during release to the registry.
+module(
+    name = "rules_ios",
+    version = "3.0.0",
```
Comment (fmeum) [inline modules/rules_ios/3.0.0/MODULE.bazel:9]: This version should match the declared module version (`3.0.0`). Are you using the Publish to BCR app? It should automatically generate a patch that does this for you.

@meteorcloudy It looks like we are missing a check that enforces module version and metadata version to be identical.

**Review by fmeum — CHANGES_REQUESTED** [review]

File: modules/rules_ios/3.0.0/MODULE.bazel (hunk 2 above)
Comment (luispadron) [reply modules/rules_ios/3.0.0/MODULE.bazel:9]: +1 to adding a check for version, we planned to use the app but when we tagged this version it wasn't configured correctly so it never ran. Is it possible to trigger it manually?

File: modules/rules_ios/3.0.0/presubmit.yml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+tasks:
+  bcr_test_module:
+    name: BCR test module
+    platform: macos
```
Comment (fmeum) [inline modules/rules_ios/3.0.0/presubmit.yml:4]: This looks incomplete now.

File: module_dot_bazel_version.patch [hunk 4]
```diff
@@ -0,0 +1,11 @@
+--- a/MODULE.bazel
```
Comment (fmeum) [inline module_dot_bazel_version.patch:1]: This needs to live under the module directory, ideally in a `patches` subdirectory.

**Comment by luispadron** [comment]
@fmeum thanks for the help here! Im going to go the route of creating a new tag, but first fixing up the `presubmit.yml` let me know if this works: https://github.com/bazel-ios/rules_ios/pull/787

We dont ship the tests, only the rule sources in our release archive.

**Pull request closed without merging**